x-pack/plugins/security_solution/public/resolver/view/panel.tsx

```diff
@@ -3,191 +3,283 @@
-import { ResolverEvent } from '../../../common/endpoint/types';
-import * as event from '../../../common/endpoint/models/event';
-import { useResolverDispatch } from './use_resolver_dispatch';
+import { useHistory } from 'react-router-dom';
+// eslint-disable-next-line import/no-nodejs-modules
+import querystring from 'querystring';
+import { EuiPanel } from '@elastic/eui';
+import { displayNameRecord } from './process_event_dot';
 import * as selectors from '../store/selectors';
+import { useResolverDispatch } from './use_resolver_dispatch';
+import * as event from '../../../common/endpoint/models/event';
+import { ResolverEvent } from '../../../common/endpoint/types';
+import { SideEffectContext } from './side_effect_context';
+import { ProcessEventListNarrowedByType } from './panels/panel_content_related_list';
+import { EventCountsForProcess } from './panels/panel_content_related_counts';
+import { ProcessDetails } from './panels/panel_content_process_detail';
+import { ProcessListWithCounts } from './panels/panel_content_process_list';
+import { RelatedEventDetail } from './panels/panel_content_related_detail';
+import { CrumbInfo } from './panels/panel_content_utilities';
 
-const HorizontalRule = memo(function HorizontalRule() {
-  return (
-    <EuiHorizontalRule
-      style={{
-        /**
-         * Cannot use `styled` to override this because the specificity of EuiHorizontalRule's
-         * CSS selectors is too high.
-         */
-        marginLeft: `-${euiVars.euiPanelPaddingModifiers.paddingMedium}`,
-        marginRight: `-${euiVars.euiPanelPaddingModifiers.paddingMedium}`,
-        /**
-         * The default width is 100%, but this should be greater.
-         */
-        width: 'auto',
-      }}
-    />
-  );
-});
-
-export const Panel = memo(function Event({ className }: { className?: string }) {
-  interface ProcessTableView {
-    name: string;
-    timestamp?: Date;
-    event: ResolverEvent;
-  }
+/**
+ * The team decided to use this table to determine which breadcrumbs/view to display:
+ *
+ * | Crumb/Table            | &crumbId                   | &crumbEvent              |
+ * | :--------------------- | :------------------------- | :----------------------  |
+ * | all processes/default  | null                       | null                     |
+ * | process detail         | entity_id of process       | null                     |
+ * | relateds count by type | entity_id of process       | 'all'                    |
+ * | relateds list 1 type   | entity_id of process       | valid related event type |
+ * | related event detail   | event_id of related event  | entity_id of process     |
+ *
+ * This component implements the strategy laid out above by determining the "right" view and doing some other housekeeping e.g. effects to keep the UI-selected node in line with what's indicated by the URL parameters.
+ *
+ * @returns {JSX.Element} The "right" table content to show based on the query params as described above
```

Comment:
Still wrapping my head around this one. So these are the concerns of this component:
1. interpreting the query params in order to decide:
  - what view to show
  - what process to show details about
  - whether any related event or related event type is selected.
2. Firing an action that informs the state about the above
3. Defines a function that can be used to update the query string w/ newly selected params
4. Renders the individual panel configurations (detail, lists, etc)

I think 1 and 2 should be done in the redux side of things, but at the very least, they should be in a separate hook (which would be in a separate file.)
For 3, i think we should move it to a separate hook, in a separate file (until we can come back and refactor it.)
I think '4' should be the only concern of this component.

Reply:
I think I agree with this approach broadly. 1 & 2 allows us to keep redux/state isolated from UA concerns (which satisfies the concern of keeping it testable/independent etc.). Combined with your suggestion yesterday of returning functions from selectors, we could "have our cake" by removing a lot of the logic/work from React and "eat it too" by keeping just the parts where we introspect or modify `window.location.history` there.